### PR TITLE
Publish package to npm

### PR DESCRIPTION
**Pull request by yuantu1919**
This is to address https://github.com/WeTrustPlatform/spring/issues/586

After this PR is merged in, to publish the package:
1. on terminal, run `npm login` to log in with your npm credential
2. run `npm publish` to publish the package.
3. Then you will see the package is successfully published as `@wetrustplatform/spring-nft-contracts`
4. Then in Spring project, we could import the package using `npm i @wetrustplatform/spring-nft-contracts`

I have tested using `npm pack` to pack the package locally and then install in Spring using `npm i <path to the wetrustplatform-spring-nft-contracts-1.0.0.tgz>` and it works as expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+  "main": ".exported.js",
   "directories": {
     "test": "test"
   },
   "scripts": {
     "lint": "solium -d contracts/ --fix",
-    "test": "solidity-coverage"
+    "test": "solidity-coverage",
+    "compile": "truffle compile",
+    "beforePublish": "truffle compile && node dist.js"
```
Comment (minheq) [inline package.json]: what is `node dist.js`

File: dist.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const glob = require('glob');
+const path = require('path');
+const fs = require('fs');
+
+// As we don't want to expose the local build folder
+// the script filters 'public' fields of the built contracts
+const contractsFiles = glob.sync(path.resolve(__dirname, './build/contracts/*.json'));
```
Comment (knd) [inline dist.js]: I don't think we need other `*.json` files except `SpringNFT.json`.

**Comment by knd** [comment]
@yuantu1919  Please document the steps how you are going to publish to NPM

File: package.json (hunk 1 above)
Comment (yuantu1919) [reply package.json]: it's to execute the dist.js file

File: package.json [hunk 3]
```diff
@@ -1,14 +1,16 @@
 {
-  "name": "spring-nft-contracts",
-  "version": "0.0.1",
-  "description": "WeTrust Spring Smart Contracts",
-  "main": "truffle-config.js",
+  "name": "@wetrustplatform/spring-nft-contracts",
+  "version": "1.0.0",
```
Comment (knd) [inline package.json]: @yuantu1919 let's keep this 0.0.1

There's no change to the implementation of nft smart contract.

Comment (yuantu1919) [reply package.json]: ok

**Pull request merged**